### PR TITLE
Use 'sts.PresignClient' for EKS cluster enrollment.

### PR DESCRIPTION
**Pull request by AntonAM**
This PR changes the way we presign `GetCallerIdentity` action, which later is used for accessing EKS cluster. We used default host for the AWS STS service, but that didn't work for all regions. Here we change to the `PresignClient` from the SDK that would take care of the host automatically.

Fixes: https://github.com/gravitational/teleport/issues/44977

Changelog: Fix EKS cluster enrollment in the Discover UI for AWS regions other than us-east-1.

### CODE REVIEW

**Review by marcoandredinis — APPROVED** [review]
I did not test this, but the changes look good

File: lib/integrations/awsoidc/eks_enroll_clusters.go [hunk 1]
```diff
@@ -299,7 +307,37 @@ func EnrollEKSClusters(ctx context.Context, log *slog.Logger, clock clockwork.Cl
...
+	addEKSHeaders := func(ctx context.Context, in middleware.BuildInput, next middleware.BuildHandler) (
+		out middleware.BuildOutput, metadata middleware.Metadata, err error,
+	) {
+		req, ok := in.Request.(*smithyhttp.Request)
+		if !ok {
+			return out, metadata, fmt.Errorf("unknown transport type %T", req)
+		}
+
+		req.Header.Add(awsHeaderClusterName, clusterName)
+		req.Header.Add(awsHeaderExpires, "60")
```
[the remark this answers is not in the document]
Comment (AntonAM) [reply lib/integrations/awsoidc/eks_enroll_clusters.go:325]: Just for compatibility reasons, this value is actually ignored, but as I understand before, 60 seconds was the maximum time that could be requested. This is what their own aws cli uses.

File: lib/integrations/awsoidc/eks_enroll_clusters.go [hunk 2]
```diff
@@ -299,7 +307,37 @@ func EnrollEKSClusters(ctx context.Context, log *slog.Logger, clock clockwork.Cl
 	return &EnrollEKSClusterResponse{Results: results}, nil
 }
 
-func enrollEKSCluster(ctx context.Context, log *slog.Logger, clock clockwork.Clock, credsProvider aws.CredentialsProvider, clt EnrollEKSCLusterClient, proxyAddr, clusterName string, req EnrollEKSClustersRequest) (string, error) {
+func presignCallerIdentityURL(ctx context.Context, stsClient *sts.Client, clusterName string) (string, error) {
+	presignClient := sts.NewPresignClient(stsClient)
+
+	// This function adds required headers for accessing an EKS cluster to the presigned URL.
```
[the remark this answers is not in the document]
Comment (AntonAM) [reply lib/integrations/awsoidc/eks_enroll_clusters.go:313]: There's not really much documentation about this, you basically need to look at the source code of their aws cli to see how it works. But I've added some comments about the headers.

File: lib/integrations/awsoidc/eks_enroll_clusters.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (AntonAM) [reply lib/integrations/awsoidc/eks_enroll_clusters.go:325]: AWS requires it to be present, it's not the cleanest API in the world 🤷

**Review by rosstimothy — APPROVED** [review]

**Pull request merged**